hpx/runtime/parcelset/parcelport.hpp

```diff
@@ -370,11 +350,10 @@ namespace hpx { namespace parcelset
         /// mutex for all of the member data
         mutable lcos::local::spinlock mtx_;
 
-        /// The handler for all incoming requests.
-        server::parcelport_queue parcels_;
+        hpx::applier::applier *applier_;
 
         /// The cache for pending parcels
-        typedef std::pair<std::vector<parcel>, std::vector<write_handler_type> >
+        typedef std::pair<std::deque<parcel>, std::deque<write_handler_type> >
```

Comment:
See here: https://github.com/STEllAR-GROUP/hpx/blob/simplify_parcel/hpx/runtime/parcelset/parcelport_impl.hpp#L554

This is related to the change of how the parcels are sent out. Instead of just swapping the pending parcels, we just pop the front and serialize it. This change was needed because whenever there is a future inside the pending parcels list which isn't ready, might hold back the remaining parcels. This ultimately might lead to a deadlock. As such, every single parcel needs to be serialized independently.